extensions/amp-story/0.1/amp-story.js

```diff
@@ -822,11 +822,16 @@ export class AmpStory extends AMP.BaseElement {
         this.updateBackground_(this.activePage_.element);
       }
     } else {
-      if (this.element.offsetWidth > this.element.offsetHeight) {
-        this.element.setAttribute('landscape', '');
-      } else {
-        this.element.removeAttribute('landscape');
-      }
+      this.vsync_.runPromise({
+        measure: () => {
+          const {offsetWidth, offsetHeight} = this.element;
+          if (offsetWidth > offsetHeight) {
+            this.element.setAttribute('landscape', '');
```

Comment:
1. `setAttribute` should be set on `mutate`.
2. No need for `runPromise`, just `run`.
3. I missed this before: I'd prefer to we have a `i-amphtml-story-landscape` classname instead of an attribute. Two reasons: a) it matches the rest of AMP better and b) keeps authors from overriding/colluding it.
```javascript
this.vsync_.run({
  measure: state  => {
    const {offsetWidth, offsetHeight} = this.element;
    state.isLandscape = offsetWidth > offsetHeight;
  },
  mutate: state => {
    this.element.classList.toggle('i-amphtml-story-landscape', state.isLandscape);
  },
}, {});
```

Reply:
oh thanks... anywhere i can read about this `vsync` more?

Reply:
Done